src/Raven.Server/Documents/Handlers/AI/Agents/ShardedAiAgentHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Threading.Tasks;
+using Raven.Server.Documents.Sharding.Handlers;
+using Raven.Server.Routing;
+
+namespace Raven.Server.Documents.Handlers.AI.Agents
+{
+    public sealed class ShardedAiAgentHandler : ShardedDatabaseRequestHandler
+    {
+        [RavenShardedAction("/databases/*/admin/ai/agent", "PUT")]
+        public Task AddOrModifyAiAgent()
+        {
+            throw new NotSupportedException("This operation is not available from a specific shard");
```

Comment:
better message and use `NotSupportedInShardingException`

Reply:
Done